### PR TITLE
fix(meta): device yaml marshal to Json  error

### PR DESCRIPTION
**Pull request by wangshihui**
when add some complex vaules to the attributes of deviceResources in the device profile, error happens. use gopkg.in/yaml.v3 instead of gopkg.in/yaml.v2

Fixes: #3682

Signed-off-by: wangshihui <wangshihui_1985@163.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [x] Yes
- [ ] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -18,6 +18,7 @@ require (
 	golang.org/x/crypto v0.0.0-20210711020723-a769d52b0f97
 	gopkg.in/eapache/queue.v1 v1.1.0
 	gopkg.in/yaml.v2 v2.4.0
```
Comment (cloudxxx8) [inline go.mod:20]: please try to remove v2 and replace all of code to v3

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

**Comment by cloudxxx8** [comment]
once you remove gopkg.in/yaml.v2, please also remove it from Attribution.txt

File: go.mod (hunk 1 above)
Comment (wangshihui) [reply go.mod]: done

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**